js/pathfinding.js

```diff
@@ -85,52 +188,28 @@ export function findPath(from, to, token, previousWaypoints) {
 		paintGridlessPathfindingDebug(pathfinder);
 		return GridlessPathfinding.findPath(pathfinder, from, to);
 	} else {
-		const cachedNodes = getCachedNodes(token);
-		const lastNode = calculatePath(from, to, cachedNodes, token, previousWaypoints);
-		if (!lastNode)
+		const cachedNodes = cache.getCachedNodes(token);
+		cache.startBackgroundCaching(token);
+		const firstNode = calculatePath(from, to, cachedNodes, token, previousWaypoints);
```

Comment:
To clear up confusion, make it clear that this is the first node in the path, and we're moving forwards.